### PR TITLE
MULE-9500: Allow specifying custom log4j config locations

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: modules/launcher/src/test/resources/log4j2-test-custom.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (marianogonzalez) [inline modules/launcher/src/test/resources/log4j2-test-custom.xml]: this file doesn't seem referenced anywhere

File: modules/launcher/src/main/java/org/mule/module/launcher/log4j2/ArtifactAwareContextSelector.java [hunk 2]
```diff
@@ -180,10 +183,30 @@ public void execute()
 
     private URI getArtifactLoggingConfig(ArtifactClassLoader muleCL)
```
Comment (marianogonzalez) [inline modules/launcher/src/main/java/org/mule/module/launcher/log4j2/ArtifactAwareContextSelector.java]: extract the entire buildContext() method (which includes this sub rutine) into a factory which you can unit test using mock application descriptors and classloaders. In that way you can test all this logic without relying on classpath discovery of test log4j config files which we know ends up in flackyness

File: modules/launcher/src/test/resources/log4j2-test-custom.xml (hunk 1 above)
Comment (elrodro83) [inline modules/launcher/src/test/resources/log4j2-test-custom.xml]: Redid the test that uses this file.

File: modules/launcher/src/main/java/org/mule/module/launcher/log4j2/ArtifactAwareContextSelector.java [hunk 3]
```diff
@@ -57,10 +47,11 @@
 class ArtifactAwareContextSelector implements ContextSelector, Disposable
 {
 
-    private static final StatusLogger logger = StatusLogger.getLogger();
+    static final StatusLogger logger = StatusLogger.getLogger();
 
     private LoggerContextCache cache = new LoggerContextCache(this, getClass().getClassLoader());
 
+    private static MuleLoggerContextFactory loggerContextFactory = new MuleLoggerContextFactory();
```
Comment (marianogonzalez) [inline modules/launcher/src/main/java/org/mule/module/launcher/log4j2/ArtifactAwareContextSelector.java]: static state is evil

**Pull request merged**